### DISCUSSION TITLE
Is it okay to call register multiple times?

### DISCUSSION THREAD

**Discussion by Moshappp** [react-hook-form/react-hook-form#10011] [Q&A]
I have this code
```
 useEffect(() => {
    register("domainList", domainList);
  }, [register, domainList]);

  useEffect(() => {
    setValue("domainList", domainList);
  }, [setValue, domainList]);
```
Since I am not rendering the list to UI but I want it to be included in React-Hook-Form validation, I registered it using effect and when the list is updated I call setValue to make React-Hook-Form aware of the latest update.

But when the list is changed the first useEffect() is called again to register domainList and if I removed the domainList from dependency array it will show an error, so is it okay to call register on specific field multiple times or there is a better way?

**Comment by paulleonartcalvo** [comment]
Even if you're not visualizing that component, you should still register it and then make it `hidden` so that it doesn't show up in the UI. Something like this:

```
      <input type="hidden" {...register('domainList')} />
```

Ref: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden
